### DISCUSSION TITLE
Feature Request: Increase queries performance on Hive by avoiding to list all files.

### DISCUSSION THREAD

**Discussion by Kuinox** [duckdb/duckdb#13614] [Ideas]
Hello,
Currently DuckDB takes a long time to list all files when running a query on a big Hive, it takes even more times when the files are on S3.
I see that duckdb runs 30 requests over 5 seconds to list 60k files, in a query that filter on the partitions matching a single file.  
This becomes way too slow when there are much more files (up to 3 Millions in my tests).  

Since the files are partitioned, and DuckDB is aware of the partition, it could be smarter when querying the files to use the filters to only list the needed files.  
S3 have Common Prefixes, on a partition like date/some_id, using this API would allow to immediately filter down the required dates.

Related issue: https://github.com/duckdb/duckdb/issues/9474

**Comment by outinspace** [comment]
I'm also interested in this optimization. I am prototyping a multi-tenant reporting engine that is partitioned by TenantId, then Date. Ideally, single-tenant queries should be able to ignore all files for other tenants. In practice, adding more tenants slows down DuckDB even if those tenants are not included in the query. It seems that DuckDB is recursively listing all files instead of listing one level at a time and filtering out partitions before listing the next level.

**Comment by soerenwolfers** [comment]
see https://github.com/duckdb/duckdb/discussions/7620
